### PR TITLE
bump dependencies and msrv

### PR DESCRIPTION
**Pull request by Humandoodlebug**
Bumps MSRV from 1.70 to 1.79 and updates dependencies.

Supercedes #224, #225, #228, #229 and #230.

### CODE REVIEW

File: src/cli/mod.rs [hunk 1]
```diff
@@ -194,15 +194,17 @@ fn generate_endpoint_subcommand(name: &str, endpoint: &Endpoint) -> clap::Comman
...
             Post => command.subcommand(
                 clap::Command::new("post").args(
-                    &endpoint
+                    endpoint
                         .fields
                         .iter()
                         .filter_map(|(name, field)| generate_post_field_arg(name, field))
                         .collect::<Vec<clap::Arg>>(),
                 ),
             ),
```
Comment (AlexMorson) [inline src/cli/mod.rs:206]: What's the reason for collecting here? As far as I can tell, `args(...)` immediately iterates over the input anyway.

**Review by AlexMorson — APPROVED** [review]
Just a couple questions, otherwise looks reasonable!

File: deny.toml [hunk 2]
```diff
@@ -1,11 +1,9 @@
 
 [advisories]
+version = 2
```
[the remark this answers is not in the document]
Comment (Humandoodlebug) [reply deny.toml:3]: Ah, indeed! I think that must have just been deprecated due to dropping support for version 1.

File: src/cli/mod.rs (hunk 1 above)
Comment (Humandoodlebug) [reply src/cli/mod.rs:206]: Indeed, this looks unnecessary. I had thought that `generate_post_field_arg` was doing more than it seems it is.

**Pull request merged**